tools/tsp-client/ci.yml

```
@@ -0,0 +1,124 @@
+# Node.js
+# Build a general Node.js project with npm.
+# Add steps that analyze code, save build artifacts, deploy, and more:
+# https://docs.microsoft.com/azure/devops/pipelines/languages/javascript
+
+trigger:
+  branches:
+    include:
+      - main
+      - feature/*
+      - release/*
+      - hotfix/*
+  paths:
+    include:
+      - tools/tsp-client
+
+pr:
+  branches:
+    include:
+      - main
+      - feature/*
+      - release/*
+      - hotfix/*
+  paths:
+    include:
+      - tools/tsp-client
+
+variables:
+  - template: ../../eng/pipelines/templates/variables/globals.yml
+  - name: NodeVersion
+    value: '18.x'
+  - name: VAR_ARTIFACT_NAME
+    value: 'drop'
+  - name: VAR_BUILD_ARTIFACT_STAGING_DIRECTORY
+    value: $(Build.ArtifactStagingDirectory)
+
+stages:
+  - stage: InstallAndBuild
+    jobs:
+      - job: Build
+        strategy:
+          matrix:
+            linux:
+              OSVmImage: 'MMSUbuntu20.04'
+              Pool: 'azsdk-pool-mms-ubuntu-2004-general'
+            mac:
+              OSVmImage: 'macos-latest'
```

```suggestion
              OSVmImage: 'macos-12'
```

We usually pin major OS versions.